providers/azure/aci.go

```diff
@@ -45,7 +47,7 @@ type AuthConfig struct {
 }
 
 // NewACIProvider creates a new ACIProvider.
-func NewACIProvider(config string, rm *manager.ResourceManager, nodeName, operatingSystem string) (*ACIProvider, error) {
+func NewACIProvider(config string, rm *manager.ResourceManager, nodeName, operatingSystem string, internalIP string, daemonEndpointPort int32) (*ACIProvider, error) {
```

Comment:
This parameter list is starting to get quite long. Maybe we should consider having things like ResourceManager passed in directly, but refactoring out all the configuration related things, (nodeName, OS, ip, daemon port, etc) out into like a ProviderConfig type. This will also prevent us from breaking all the providers when we add new params.

Reply:
@erikstmartin do you mind if we do this in a separate PR so that this PR can focus on just enabling logging?

Reply:
@erikstmartin are you fine with merge the PR and refactor the parameters later?